test/integration/cds_integration_test.cc

```diff
@@ -194,8 +198,16 @@ TEST_P(CdsIntegrationTest, CdsClusterUpDownUp) {
 // Make sure that clusters won't create new connections on teardown.
 TEST_P(CdsIntegrationTest, CdsClusterTeardownWhileConnecting) {
   initialize();
+  // This line ensures that the ClusterInfoImpl is created already.
   test_server_->waitForCounterGe("cluster_manager.cluster_added", 1);
-  test_server_->waitForCounterExists("cluster.cluster_1.upstream_cx_total");
+  if (this->enableLazyInitStats()) {
+    EXPECT_EQ(test_server_->gauge("cluster.cluster_1.ClusterTrafficStats.inited")->value(), 0);
```

Comment:
make temps for the gauge and counter rather than doing the string lookup twice.

I think also you should get the GaugeSharedPtr either way, and ASSERT_NE(nullptr, init_gauge) of lazy-init is enabled, and ASSERT_EQ(nullptr, init_gague) otherwise.